### ISSUE TITLE
Station magnitude residual of zero not read from QuakeML

### ISSUE THREAD

**Issue by nackerley** [obspy/obspy#1621]
Using ObsPy 1.02 and Python 3.5 installed via Anaconda on Windows.

Try the following  with [test.zip](https://github.com/obspy/obspy/files/646936/test.zip) unzipped to the working directory:

```python
from obspy import read_events
catalog = read_events('test.xml')
print(catalog.events[0].magnitudes[0].station_magnitude_contributions[2].residual)
catalog.write('test2.xml', 'QuakeML')
```

The problem is that station magnitude residual of 0.0 is read and printed, but not written to test2.xml.
It seems that a residual of 0.0 in the third StationMagnitudeContribution is explicitly ignored on writing; you can change the residual to another value and it will write fine.

I expect a residual of 0.0 to be just as valid as any other value. I would expect a value of None to indicate that the residual is not known or not recorded.

**Comment by krischer** [comment]
I don't have the time right now but if someone want to fix it - this line here as to check "is not None":

https://github.com/obspy/obspy/blob/9fb860d9868ae5c595ad88e55f317bf84a2d1256/obspy/io/quakeml/core.py#L1164

**Comment by krischer** [comment]
There might be a couple of other places where this might have to be done.

**Comment by nackerley** [comment]
I think Event.region may need the same treatment.

**Comment by nackerley** [comment]
I've verified that your proposed fix does what we want it to @krischer, by crudely editing the file in my site-packages. Ugly. For want of a C++ compiler on the Windows box I normally develop on I would have forked, fixed and pushed the fix. I hope to be able to soon.

**Comment by megies** [comment]
Fixed in #1625.
